### ISSUE TITLE
Concat of empty & non-empty dataframe causes PanicException "Column lengths not equal"

### ISSUE THREAD

**Issue by phil-zxx** [pola-rs/polars#19799]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
import datetime as dt

df       = pl.DataFrame()
df_extra = pl.DataFrame([{'name': 'ABC', 'date': dt.date(2024, 11, 1)}])
df       = pl.concat([df, df_extra])

print(len(df))
print(df)
```

### Issue description

Using `pl.concat` with an empty and non-empty dataframe causes a `PanicException`.

### Expected behavior

In polars version `1.9.0` (or earlier), the two `print` statements display
```
>> print(len(df))
1
```
and
```
>> print(df)
shape: (1, 2)
┌──────┬────────────┐
│ name ┆ date       │
│ ---  ┆ ---        │
│ str  ┆ date       │
╞══════╪════════════╡
│ ABC  ┆ 2024-11-01 │
└──────┴────────────┘
```

But in polars version `1.12.0` I am now getting
```
>> print(len(df))
0
```
which seems incorrect (should probably be 1). And then
```
>> print(df)
File site-packages\polars\dataframe\frame.py:1178, in DataFrame.__str__(self)
   1177 def __str__(self) -> str:
-> 1178     return self._df.as_str()

PanicException: The column lengths in the DataFrame are not equal.
```

_(note, I cannot test polars `1.13.0` or higher right now, as `1.12.0` is the latest in our artifactory)_

### Installed versions

<details>

```
--------Version info---------
Polars:              1.12.0
Index type:          UInt32
Platform:            Windows-11-10.0.26120-SP0
Python:              3.12.4 (tags/v3.12.4:8e8a4ba, Jun  6 2024, 19:30:16) [MSC v.1940 64 bit (AMD64)]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               <not installed>
gevent               <not installed>
great_tables         <not installed>
matplotlib           3.9.2
nest_asyncio         1.6.0
numpy                2.1.1
openpyxl             <not installed>
pandas               2.2.2
pyarrow              16.1.0
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                <not installed>
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by cmdlineluser** [comment]
The panic was fixed in 1.13.0 

- https://github.com/pola-rs/polars/pull/19642

**Comment by phil-zxx** [comment]
Oh, amazing, thank you.

**Comment by alexander-beedie** [comment]
Closed by #19642.
